### PR TITLE
Feature/deploy

### PR DESCRIPTION
**Pull request by elisehuard**
Deployment of kixi.heimdall on marathon-mesos

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@ pom.xml.asc
 /.lein-*
 /.nrepl-port
 *.pem
+prod.auth-conf.edn
+*.json
```
Comment (otfrom) [inline .gitignore]: All json everywhere or just in some directories?

File: dev/user.clj [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
   (let [new-system (resolve 'kixi.heimdall.system/system)]
     (alter-var-root #'kixi.heimdall.application/system
-                    (constantly (new-system)))))
+                    (constantly (new-system :development)))))
```
Comment (otfrom) [inline dev/user.clj]: always :development?

File: heimdall.json.template [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          {
+              "containerPort": 5001,
+              "hostPort": 10011,
+              "protocol": "tcp"
+          }
+      ]
+    }
+  },
+  "id": "heimdall",
+  "cpus": 1,
```
Comment (otfrom) [inline heimdall.json.template]: could probably get away w/something like 0.2 cpus.

File: heimdall.json.template [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "mem": 1024,
```
Comment (otfrom) [inline heimdall.json.template]: can it use less ram? I'm thinking we'd want it to use fewer resources so we can have multiple instances running across the cluster w/o too much trouble.

File: resources/conf.edn [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{:cassandra-session {:hosts #profile {:development ["localhost"]
+                                      :production ["node-0.cassandra.mesos"
+                                                   "node-1.cassandra.mesos"
+                                                   "node-2.cassandra.mesos"]}
```
Comment (otfrom) [inline resources/conf.edn]: annoying we have to specify all the hosts (as there could potentially be a lot if things grew quite a bit)

File: src/kixi/heimdall/handler.clj [hunk 6]
```diff
@@ -18,8 +18,8 @@
   (edn/read-string (slurp f)))
 
 (def auth-config
-  (get-config (or (util/file-exists? (System/getProperty "user.home") ".heimdall.auth-conf.edn")
-                  (io/resource (:auth-conf env)))))
+  (util/get-config (or (util/file-exists? (System/getProperty "user.home") ".heimdall.auth-conf.edn")
```
Comment (otfrom) [inline src/kixi/heimdall/handler.clj]: should get-config live in kixi.heimdall.config meaning that util could be dropped? (I'm generally against util workspaces for things that aren't truly domainless)

File: src/kixi/heimdall/util.clj [hunk 7]
```diff
@@ -1,7 +1,12 @@
 (ns kixi.heimdall.util
   (:require [clj-time.format :as tf]
             [clj-time.core :as t]
-            [clojure.java.io :as io]))
+            [clojure.java.io :as io]
+            [clojure.edn :as edn]))
+
+(defn get-config
+  [f]
+  (edn/read-string (slurp f)))
```
Comment (otfrom) [inline src/kixi/heimdall/util.clj]: get-config could go in config namespace. 

db-now below could go into a time one (there will probably be others that will end up in there too)

replacer, underscore->hyphen and hyphen->underscore could go into a conformer or coerecer namespace

File: dev/user.clj (hunk 2 above)
Comment (elisehuard) [inline dev/user.clj]: this is for use in the repl.  In prod we pass in a parameter to the jar for the profile (see bootstrap.clj)

**Comment by otfrom** [comment]
happy to :shipit: after those are fixed

File: resources/conf.edn (hunk 5 above)
Comment (elisehuard) [inline resources/conf.edn]: all the hosts we want it to access, yes. I imagine that it doesn't ahve to be ALL of them if that becomes a problem.

File: .gitignore [hunk 8]
```diff
@@ -9,3 +9,5 @@ pom.xml.asc
 /.lein-*
 /.nrepl-port
 *.pem
+heirmdall.json
```
Comment (otfrom) [inline .gitignore]: heiRmdall?

**Comment by otfrom** [comment]
I'd still like to see the util ns disappear at some point, but :shipit:

**Pull request merged**